### ISSUE TITLE
Outlives obligations aren't normalized before processing

### ISSUE THREAD

**Issue by compiler-errors** [rust-lang/trait-system-refactor-initiative#61]
This code should compile:

```rust
trait Tr<'a> {
    type Assoc;
}

fn outlives<'o, T: 'o>() {}

fn foo<'a, 'b, T: Tr<'a, Assoc = ()>>() {
    outlives::<'b, T::Assoc>();
}
```

Similar to #59, but in the other direction (should pass, but fails).

**Comment by compiler-errors** [comment]
This is not fixed by rust-lang/rust#119101, MIR borrowck doesn't normalize type-outlives yet. That will happen soon.

**Comment by compiler-errors** [comment]
This was fixed in https://github.com/rust-lang/rust/pull/120513
